dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/AddFilesResult.java

```diff
@@ -0,0 +1,34 @@
+package edu.harvard.iq.dataverse.dataset;
+
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+
+public class AddFilesResult {
+
+    private Dataset dataset;
+    
+    private int notSavedFilesCount;
+    
+    private boolean hasProvenanceErrors;
+
+    // -------------------- CONSTRUCTORS --------------------
+    
+    public AddFilesResult(Dataset dataset, int notSavedFilesCount, boolean hasProvenanceErrors) {
+        this.dataset = dataset;
+        this.notSavedFilesCount = notSavedFilesCount;
+        this.hasProvenanceErrors = hasProvenanceErrors;
+    }
+
+    // -------------------- GETTERS --------------------
+    
+    public Dataset getDataset() {
+        return dataset;
+    }
+
+    public int getNotSavedFilesCount() {
+        return notSavedFilesCount;
+    }
+
+    public boolean isHasProvenanceErrors() {
```

Comment:
Imo it would be beneficial to explain what is Provenance in dataverse.

Reply:
Here? I don't think this is the place to explain the whole mechanism. I will add some comment explaining what exactly refers to

Reply:
Hmm yeah good point.